docs/library/espnow.rst

```diff
@@ -0,0 +1,917 @@
+For example::
+
+    import network
+
+    sta = network.WLAN(network.STA_IF)
+    sta.active(True)
+    sta.disconnect()    # For ESP8266
+
+**Note:** The ESP8266 has a *feature* that causes it to automatically reconnect
+to the last wifi Access Point when set `active(True)<network.WLAN.active>` (even
+after reboot/reset). This reduces the reliability of receiving ESP-NOW messages
+(see `ESPNow and Wifi Operation`_). You can avoid this by calling
+`disconnect()<network.WLAN.disconnect>` after
+`active(True)<network.WLAN.active>`.
+
+.. method:: ESPNow.send(mac, msg[, sync])
+            ESPNow.send(msg)   (ESP32 only)
+
+    Send the data contained in ``msg`` to the peer with given network ``mac``
+    address. In the second form, ``mac=None`` and ``sync=True``. The peer must
+    be registered with `ESPNow.add_peer()<ESPNow.add_peer()>` before the
+    message can be sent.
+
+    .. data:: Arguments:
+
+      - ``mac``: byte string exactly ``espnow.ADDR_LEN`` (6 bytes) long or
+        ``None``. If ``mac`` is ``None`` (ESP32 only) the message will be sent
+        to all registered peers, except any broadcast or multicast MAC
+        addresses.
+
+      - ``msg``: string or byte-string up to ``espnow.MAX_DATA_LEN`` (250)
+        bytes long.
+
+      - ``sync``:
+
+        - ``True``: (default) send ``msg`` to the peer(s) and wait for a
+          response (or not).
+
+        - ``False`` send ``msg`` and return immediately. Responses from the
+          peers will be discarded.
+
+    .. data:: Returns:
+
+      ``True`` if ``sync=False`` or if ``sync=True`` and *all* peers respond,
+      else ``False``.
+
+    .. data:: Raises:
+
+      - ``OSError(num, "ESP_ERR_ESPNOW_NOT_INIT")`` if not initialised.
+      - ``OSError(num, "ESP_ERR_ESPNOW_NOT_FOUND")`` if peer is not registered.
+      - ``OSError(num, "ESP_ERR_ESPNOW_IF")`` the wifi interface is not
+        `active()<network.WLAN.active>`.
+      - ``OSError(num, "ESP_ERR_ESPNOW_NO_MEM")`` internal ESP-NOW buffers are
+        full.
+      - ``ValueError()`` on invalid values for the parameters.
+
+    **Note**: A peer will respond with success if its wifi interface is
+    `active()<network.WLAN.active>` and set to the same channel as the sender,
+    regardless of whether it has initialised it's ESP-Now system or is
+    actively listening for ESP-Now traffic (see the Espressif ESP-Now docs).
+
+.. method:: ESPNow.recv([timeout])
+
+    Wait for an incoming message and return the ``mac`` adress of the peer and
+    the message. **Note**: It is **not** necessary to register a peer (using
+    `add_peer()<ESPNow.add_peer()>`) to receive a message from that peer.
+
+    .. data:: Arguments:
+
+        ``timeout``: *(Optional)* If provided and not `None`, sets a timeout (in
+        milliseconds) for the read. The default timeout (5 minutes) is set using
+        `ESPNow.config()`. If ``timeout`` is less than zero, then wait forever.
```

Comment:
This seems to indicate that there are four possibilities for the `*timeout*` argument:
- `None`: no timeout?
- `== 0`: no timeout
- `> 0`: a timeout in milliseconds
- `< 0`: wait forever

Is `None` the same as "no timeout" or "wait forever"?  Is the `None` option even needed?

Also, should this argument be called `timeout_ms`, to make it clear of the units?

(Same comments apply to all other methods here that take a timeout argument, and to the code that implements this, to make sure it follows the docs.)

Reply:
I have modified code, docs and tests to rename "timeout" to "timeout_ms".

The `None` value (or omitted) means use the value set in `config()` (or the default).

I need this so that I can support passing the default value in the espnow.py wrapper down to the _espnow C module:
```py
    def irecv(self, timeout_ms=None):
        n = self.recvinto(self._data, timeout_ms)
        return self._data if n else self._none_tuple
```

Reply:
Your changes here look good.